soc: esp32c2: Added support to ESP32C2 and ESP8684

raffarost opened the pull request:
Added support to ESP32C2 and ESP8684 (same core).

Initial features:

- Simple boot support
- Basic drivers:
    - Clock control
    - UART
    - GPIO
    - Watchdog

Commit:
boards: esp8684_devkitm: Add support

Add support for board esp8684_devkitm

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
west.yml: Update for ESP32C2 and ESP8684

HAL update for ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

gmarull reviewed (changes requested):
please split combo commits like e56ba23ce2e09101ca1ae511675e5982713d311e properly to ease review.

Commit:
soc: esp32c2: Add support to ESP32C2 and ESP8684

Files for SoC support: ESP32C2 and ESP8684 (same core).
Basic device tree configuration.

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: clock_control: esp32c2: Add support

Support for ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: intc: esp32c2: Added support

Added support for ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

uLipe reviewed (changes requested):
In general LGTM, my only concern are the soc support and dts addition commits, I think they would be combined into a single one since no SoC support comes with dts or soc code support, so I suggest to combine:

91cc4a761a23bb6a0e2518272d7c9dc11d4ca14d

and

91cc4a761a23bb6a0e2518272d7c9dc11d4ca14d

into a single commit in this patchset.

gmarull reviewed (dismissed):
please, do not create combo commits like c941736f52f71f07f19de472c2cf32d21db84c86

Commit:
drivers: pinctrl: esp32c2: Add support

Add pinctrl support to ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: gpio: esp32c2: Add support

Add gpio support to ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: counter: esp32c2: Add support

Add counter support to ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: entropy: esp32c2: Add support

Add entropy support to ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: hwinfo: esp32c2: Add support

Add hwinfo support to ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: uart: esp32c2: Add support

Add UART support to ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: timer: esp32c2: Add support

Add timer support to ESP32C2 and ESP8684

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

marekmatej commented on boards/common/esp32.board.cmake:
```diff
@@ -13,4 +13,4 @@ board_runner_args(openocd  --gdb-init "thb main")
 assert(ESP_IDF_PATH "ESP_IDF_PATH is not set")
 
-board_finalize_runner_args(esp32 "--esp-idf-path=${ESP_IDF_PATH}")
+board_runner_args(esp32 "--esp-idf-path=${ESP_IDF_PATH}")
```
Please do not mix this with new board support PRs.

marekmatej reviewed (changes requested):
Split generic stuff from the particular board support.

marekmatej commented:
I believe splitting the support for ESP32-C2 and ESP-8684 into separate PRs would help to speed up the review process! Regardless the fact they share the same features or not.

raffarost commented:
> I believe splitting the support for ESP32-C2 and ESP-8684 into separate PRs would help to speed up the review process! Regardless the fact they share the same features or not.

The chip is the same, only difference being ESP8684 has in-chip flash. Functionally they are the same for all contents in the PR

marekmatej commented on scripts/west_commands/runners/esp32.py:
```diff
@@ -62,6 +62,8 @@ def do_add_parser(cls, parser):
                             help='serial port to flash')
         parser.add_argument('--esp-baud-rate', default='921600',
                             help='serial baud rate, default 921600')
+        parser.add_argument('--esp-monitor-baud', default='115200',
```
can you please move this change to a separate PR as it is not related to the C2 support?

raffarost replied:
This is actually for C2. The standard baud rate is 74880. so we need to pass this parameter to `monitor`

sylvioalves commented on soc/espressif/esp32c2/Kconfig:
```diff
@@ -0,0 +1,41 @@
+	select CLOCK_CONTROL
+	select PINCTRL
+	select RISCV_ISA_RV32I
+	select RISCV_ISA_EXT_A
```
I think C2 has no Atomic set. PLease confirm and remove `select RISCV_ISA_EXT_A`.

sylvioalves commented on soc/espressif/esp32c2/Kconfig:
```diff
@@ -0,0 +1,41 @@
+
+if SOC_SERIES_ESP32C2
+
+config ESP32_PHY_MAX_WIFI_TX_POWER
```
This (and other) has been moved into `Kconfig.wifi`. Not sure if merged already.

sylvioalves commented on soc/espressif/esp32c2/soc.h:
```diff
@@ -0,0 +1,56 @@
+
+void __esp_platform_start(void);
+
+static inline uint32_t esp_core_id(void)
```
Can we remove this? Either use Zephyr's arch API or cpu_ll.h if needed.

raffarost replied:
This is used in a couple of drivers and is present in all other SoC's.
Should we do a separate PR?

marekmatej commented:
@nashif could you take a look?
